luxonis_ml/tracker/tracker.py

```diff
@@ -500,6 +500,30 @@ def upload_artifact(
                 )  # Stores details for retrying later
                 self.log_stored_logs_to_mlflow()
 
+    @rank_zero_only
+    def log_matrix(
+        self, matrix: np.ndarray, name: str = "confusion_matrix"
+    ) -> None:
+        """Logs a confusion matrix as a JSON artifact by flattening the
+        matrix and saving its shape.
+
+        @type matrix: np.ndarray
+        @param matrix: The confusion matrix to log.
+        @type name: str
+        @param name: The name of the artifact.
+        """
+        matrix_data = {
+            "flat_array": matrix.flatten().tolist(),
+            "shape": matrix.shape,
+        }
+        tmp_file_path = f"{name}.json"
```

Comment:
@kozlov721 But then, would it have a random name?

Reply:
In that case, a random name would be saved in MLflow, as it would simply copy the local path.

Reply:
You can use the `name` attribute of `upload_artifact` to override the remote name.

Reply:
This would set the `remote_path` in the `fs.put_file` but for `mlflow` this variable is not used because of limitations [here](https://github.com/luxonis/luxonis-ml/blob/7332f5495ae8e63576d5271c89fbd2959070c2c5/luxonis_ml/utils/filesystem.py#L214)